### ISSUE TITLE
Set explicit alias, store and key password for pkcs 12 keystores

### ISSUE THREAD

**Issue by FlyingSheepOnSailfish** [FiloSottile/mkcert#316]
mkcert can be configured to create a pkcs#12 keystore. However:
* the alias is hardcoded to 1
* the keystore password is hardcoded to “changeit”
* the key password is hardcoded to “changeit”

It should be possible to set these 3 properties explicitly e.g.

```
mkcert -pkcs12 -p12-file acme_demo.p12 \
       -alias acme_demo \
       -storepass t0ps3cret! \
       -keypass C0nfiden!al \
       acme.demo localhost 127.0.0.1
```

Why? This would create a valid keystore in one step, that could be directly used by a Java Application Server. Yes I can take the current  p12 file from mkcert, and use Java keytool or openssl to change the passwords, but those are extra steps.

A one-line, one-tool solution would be cool!

Both mkcert -help and this issue https://github.com/FiloSottile/mkcert/issues/20 suggest that the pkcs#12 is a legacy / dying format. I wonder if that is a golang perception?  As of Java 9, pkcs#12 is the default keystore format, replacing the java jks format. That suggests that pkcs#12 is alive and kicking. I agree though that it is a nasty format, but it is not dying!

My goal is to be able to create development keystores as easily as possible, that contain proper valid certificates accepted by Firefox, Chrome, Edge etc., thus enabling developers to “do https properly” from day one.

**Comment by FiloSottile** [comment]
Thank you for opening this issue and in particular for providing context around the ecosystem usage of PKCS#12. mkcert is not meant to be a one-tool solution to anything certificate related. It's meant to be a simple, easy tool that does well one thing that many people need. If we can serve edge cases by being compatible with other tools, for example by generating something that keytool can operate on, that's what we want. It lets us take a tiny fraction of the complexity, but still serve our primary purpose. If for example tomorrow Java grows a new option to structure keystores, we don't have to do anything if people are already using keytool to massage what we generate into exactly what they need. I hope this explains my reticence to implement more PKCS#12 options!

**Comment by FlyingSheepOnSailfish** [comment]
I admire that you wish to keep mkcert focused. That is good.

The ability to set an explicit password is not really doing any more, as you already set a hardcoded password, it is doing the same thing a little bit cleaner.

Last night I forked mkcert, and made the change to set the keystore password in my fork. I only had to change / add a few lines of code: almost a one-liner fix. Thankfully your go code was clear even to a java guy like me!

I added only one new command line flag:

	-p12-password
		Explicitly set the password of the PKCS #12 file,
		the same password is used for both the keystore and the keypair.
		Empty passwords are NOT allowed.


If this flag is not set, behaviour is unchanged.

I will create a pull request.
